docs/implementation_design/packages_plugins.md

```diff
@@ -1 +1,14 @@
 # Packages and plugins
+In GoShimmer, new features are added through the [plugin system](plugin.md), however plugin objects should only expose the logic that is implemented in a different package.
+The plugin system works as an adapter that can be used to easily expose complex logic through a common interface. 
+It's really useful in a prototype software like GoShimmer, because it's possible to easily switch between different implementations just by using different plugin, without having to
```

Comment:
```suggestion
It's really useful in a prototype software like GoShimmer, because it's possible to easily switch between different implementations just by using a different plugin, without having to
```